### PR TITLE
fix(discover) Only send project ids we need

### PR DESCRIPTION
**Pull request by evanh**
Update the automatic fields/conditions to factor in if the user is filtering to
a particular project, and only send to snuba the project IDs that the user is
filtering for.

### CODE REVIEW

**Review by dashed — DISMISSED** [review]

**Review by wmak — COMMENTED** [review]
I think this broke event_facets:
```
Traceback (most recent call last):
  File "/Users/wmak/development/sentry/src/sentry/api/base.py", line 89, in handle_exception
    response = super(Endpoint, self).handle_exception(exc)
  File "/Users/wmak/development/getsentry/.venv/lib/python2.7/site-packages/rest_framework/views.py", line 449, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/wmak/development/sentry/src/sentry/api/base.py", line 196, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/wmak/development/sentry/src/sentry/api/endpoints/organization_events_facets.py", line 34, in get
    referrer="api.organization-events-facets.top-tags",
  File "/Users/wmak/development/sentry/src/sentry/snuba/discover.py", line 1003, in get_facets
    sample = len(snuba_filter.filter_keys["project_id"]) > 2
KeyError: 'project_id'
```

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -889,8 +894,11 @@ def to_list(value):
         for key in ("start", "end"):
             kwargs[key] = params.get(key, None)
         # OrganizationEndpoint.get_filter() uses project_id, but eventstore.Filter uses project_ids
-        if "project_id" in params:
-            kwargs["project_ids"] = params["project_id"]
+        if not project_to_filter and "project_id" in params:
+            if project_to_filter:
```
Comment (markstory) [inline src/sentry/api/event_search.py:898]: When would this be true if project_to_filter is falsey on the line above?

**Comment by evanh** [comment]
Boooooo event-facets tests for missing that. Thanks for noticing that @wmak .

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (evanh) [reply src/sentry/api/event_search.py:898]: My bad, I didn't refactor this correctly.

**Review by markstory — APPROVED** [review]

**Pull request merged**